### ISSUE TITLE
Email links in text/plain ending in "." - break links

### ISSUE THREAD

**Issue by flohoff** [openstreetmap/openstreetmap-website#1844]
Hi,

i am reading my emails with mutt in a terminal emulator and most(all?) OSM notification mails for Changeset comments etc contain the link ended with a "." for a correct sentence ending. The problem is that the default url regexp for all terminal emulators takes the dot "." with them.

All emails with multipart/alternative in their text/plain part should have a space between a url and the following text/template parts that urls stay detectable by terminal emulators:

Part of this problem has been discussed in https://github.com/openstreetmap/openstreetmap-website/issues/995

Here an example from an email i just got:

    More details about the changeset can be found at https://www.openstreetmap.org/changeset/58362330.
    
    To unsubscribe from updates to this changeset, visit https://www.openstreetmap.org/changeset/58362330 and click "Unsubscribe".

**Comment by tomhughes** [comment]
As discussed in #995 there is really no good solution for this - it is always going to be hit or miss whether any particular client manages to guess how to extract a link from plain text which is why we include an HTML version of the message where the boundaries are clear.

Short of rewriting sentences to avoid having links at the end, which may be very clumsy and in any case will be subject to breakage in each translation if translators move them around do you actually have a suggested solution?

**Comment by flohoff** [comment]
The point is that "." is a valid character for a URL - So ending a URL with a "." is valid. There can be no automated system which knows if the "." belongs to the URL or not. Its just us - the human beeings - looking at the osm urls beeing able to distinguish. The only valid character to end a url is a space because that is actually an invalid character (would need to be encoded as %20)

**Comment by tomhughes** [comment]
Yes but it's virtually impossible for us to guarantee a space after a URL for several reasons.

Firstly because it may just not make sense in terms of the sentence structure and secondly because even if I change all the english text to avoid having URLs at the end of sentences I have no way to ensure that translators won't undo that in other translations. There's no practical way to prevent that happening when the translations are crowdsourced.

**Comment by joernheissler** [comment]
A few suggestions on how to so solve this:
* Remove the trailing dot, even though this might be considered a grammar mistake: ``Reply or learn more about the note at https://www.openstreetmap.org/note/4153625``
* Rewrite the sentences to use a colon: ``Reply or learn more about the note at: https://www.openstreetmap.org/note/4153625``
* Enclose the link in angle brackets: ``Reply or learn more about the note at <https://www.openstreetmap.org/note/4153625>.``
* Modify the router on the website to ignore the trailing dot in links: <https://www.openstreetmap.org/note/4153625.>

**Comment by AntonKhorev** [comment]
Only this option is translation-proof, if the url is wrapped before the substitution:

> Reply or learn more about the note at \<https://www.openstreetmap.org/note/4153625>.

But it's not client-proof, as you can see here on Github.
